src/sagemaker_xgboost_container/callback.py

```diff
@@ -37,9 +17,13 @@ def add_debugging(callbacks, hyperparameters, train_dmatrix,
                              instead of default config file.
     """
     try:
-        hook = SmeDebugHook(json_config_path, hyperparameters, train_dmatrix, val_dmatrix)
+        hook = Hook.hook_from_config(json_config_path)
+        hook.hyperparameters = hyperparameters
+        hook.train_data = train_dmatrix
+        if val_dmatrix is not None:
+            hook.validation_data = val_dmatrix
+        callbacks.append(hook)
         logging.info("Debug hook created from config")
     except Exception as e:
         logging.debug("Failed to create debug hook", e)
-    else:
```

Comment:
Nit - this else is now redundant.

Reply:
umm so currently I am not using it right?

Reply:
Why keep unused code? I meant removing this else altogether would work just the same.

Reply:
Ak it's my bad. I didn't notice that return was moved under try. Fyi- you don't need the `return` either.